### PR TITLE
fix(training): honor TrainSpec.learning_rate in lerobot_local (None now means the backend default)

### LINKED ISSUE #1020: TrainSpec.learning_rate is silently ignored by the lerobot_local trainer (every other backend honors it)
## Summary

`TrainSpec` exposes `learning_rate`, the `train_policy` tool forwards it as a first-class parameter, and four of the five training backends honor it — but the `lerobot_local` trainer never wires it into the config it builds, so the policy's training-preset LR is always used. A user who sets `learning_rate` is silently misled.

## Steps to reproduce

    from strands_robots.training import TrainSpec, create_trainer
    spec = TrainSpec(dataset_root=..., output_dir=..., learning_rate=5e-5,
                     extra={"policy_type": "act"})
    cfg = create_trainer("lerobot_local").build_config(spec)
    print(cfg.policy.optimizer_lr)   # -> 1e-05 (the ACT preset), not 5e-05

Run today on main: requested `5e-05` → `cfg.policy.optimizer_lr = 1e-05`, and `build_command()` emits no `--policy.optimizer_lr` flag either.

## Actual behavior

Confirmed on a real GPU run (two-phase ACT training, base + post-tune, Colab A100): both phases logged `lr:1.0e-05` (the ACT preset) regardless of the `TrainSpec.learning_rate` value.

## Root cause / asymmetry

- `training/lerobot.py` contains no reference to `learning_rate` — neither `build_config()` (typed path) nor `build_command()` (argv-parity path) maps it.
- Every other backend uses the field: PPO (`rl/ppo.py`) and FastSAC (`rl/fast_sac.py`) bind it to their Adam optimizers; GR00T forwards `--learning_rate`; Cosmos forwards `optimizer.lr`.
- The agent-facing `train_policy` tool exposes `learning_rate` as a parameter and passes it into the spec — so the agent path advertises a knob the lerobot backend ignores.
- A working escape hatch exists and is even exercised by the test suite (`extra={"policy.optimizer_lr": ...}`), which makes the ignored first-class field the odd one out.
- In fact `training/lerobot.py` reads **every other applicable `TrainSpec` field** (dataset_root / dataset_repo_id / streaming / base_model / output_dir / steps / global_batch_size / save_freq / num_gpus / num_nodes / resume / seed / method / lora_* / tune / val_episodes) — `learning_rate` is the only universal field with zero references in the file.

## Why this isn't the documented tolerance rule

`TrainSpec`'s docstring does let backends "read the fields they support and ignore the rest" — but fields with designed partial support say so on the field itself (`seed`: "best-effort; not all backends expose it"; `streaming`: "LeRobot-only"; `tune`: "Ignored by backends that don't" expose the toggles). `learning_rate` carries no such caveat, sits in the universal block, and is advertised by the `train_policy` tool — this reads as an accidental gap, not designed partial support. (A field-by-backend sweep found sibling gaps only in the Cosmos3 trainer — filing those separately.)

## Expected behavior

An explicit `learning_rate` reaches lerobot's optimizer config (`policy.optimizer_lr`, which the policy's `get_optimizer_preset()` reads) — or is rejected loudly. Silently ignoring it is the one behavior that can't be right; same class as #897 / #968 / #979 (make silent failure loud).

One design note: the field currently defaults to `1e-4`, so wiring it unconditionally would change every *default* lerobot run (the ACT preset is 1e-5). The natural shape is the `seed` pattern the spec already uses: default `None` = "the backend's own default", an explicit value = honored everywhere.

## Environment

main @eb48cdd · lerobot 0.5.1 · confirmed on Colab A100 (real run) and locally (config-level, Python 3.12)

### PR DESCRIPTION
**Pull request by mertkrgl**
Fixes #1020.

`TrainSpec.learning_rate` was honored by four of the five backends (PPO / FastSAC bind it to their optimizers, GR00T / Cosmos forward it) and exposed as a first-class `train_policy` tool parameter — but the lerobot trainer never mapped it (the only universal spec field `training/lerobot.py` doesn't read), so the policy's training-preset LR always won and a user-set value was silently ignored. Confirmed on a real base + post-tune ACT run (A100): both phases logged `lr:1.0e-05` regardless of the spec.

The fix follows the spec's own `seed` pattern:

- `TrainSpec.learning_rate: float | None = None` — `None` means "the backend's own default". Unset behavior is unchanged in **every** backend: PPO / FastSAC / GR00T / Cosmos fall back to `DEFAULT_LEARNING_RATE = 1e-4` (the previous field default), lerobot keeps the policy preset.
- An explicit value now reaches lerobot on both build paths: `build_config()` sets `cfg.policy.optimizer_lr` (which `get_optimizer_preset()` reads) and `build_command()` emits `--policy.optimizer_lr=...` — and a policy type without an `optimizer_lr` field raises a clear `ValueError` instead of ignoring the request. (The spec's tolerance rule protects fields a backend *doesn't use*; an explicitly requested value it *can't honor* shouldn't vanish.)
- The `train_policy` tool signature follows (`learning_rate: float | None = None`; the parameter name is unchanged, so the agent action contract is intact).

Tests: 2 new in `tests/training/test_lerobot.py` (an explicit LR reaches `policy.optimizer_lr` and the argv mirror; the default `None` keeps the ACT preset and emits no flag). Existing GR00T / Cosmos / RL tests pass unchanged — the `1e-4` fallback preserves their previous rendering exactly.

Compat note (also in the CHANGELOG): the field default changed `1e-4` → `None`. No runtime behavior changes for unset specs; the griffe breaking-change report will flag the default change — that's this change, by design. If you'd rather keep `float = 1e-4` and only warn from `validate()` that lerobot ignores the field, happy to switch — the trade-off is that an explicit 1e-4 then can't be expressed for lerobot.

Found tuning a two-phase (base → post-tune) ACT training for an SO-101 dataset on Colab; the silent preset LR is why both phases trained at 1e-5.

### CODE REVIEW

**Comment by cagataycali** [comment]
Closing as a duplicate of #1022, which fixes the same issue (#1020) with the identical `learning_rate: float | None = None` opt-in design and is already approved with CI green (auto-merge armed). #1022 additionally pins the `RLTrainSpec` default to a concrete `1e-4` (RL trainers have no policy preset to defer to) and carries the wider regression set (explicit-LR-reaches-`policy.optimizer_lr`, argv mirror, `None`-keeps-preset, no-flag-emitted, and the `ValueError` on a policy without an `optimizer_lr` field). The analysis here matches and reinforces that fix. Thanks for the parallel investigation.
